src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1637,7 +1737,12 @@ bool AdaptDispatch::SoftReset()
     }
     if (success)
     {
-        success = DesignateCharset(DispatchTypes::VTCharacterSets::USASCII); // Default Charset
+        _termOutput = {}; // Reset all character set designations.
+        if (_initialCodePage != 0)
+        {
+            // Restore initial code page if previously changed by a DOCS sequence.
+            success = _pConApi->SetConsoleOutputCP(_initialCodePage);
+        }
```

Comment:
Should we clear `_initialCodePage` so the next `DCS` stores it?

(nit: should we use `std::optional` instead of magic-value-0?)

Reply:
Yeah, that's a good idea. I've done that now. It make the `GetConsoleOutputCP` call a bit of a pain, but hopefully we'll clean that up one day when we stop using references for return values.

Reply:
One day, one day :smile: thanks